Update GDScript highlighting for Godot 4

sockeye-d opened the pull request:
This removes `export` and related keywords from the list of keywords and adds a new rule for highlighting annotations

Andriamanitra commented:
I would rather keep old keywords around because someone could be editing code written for an old Godot version.

sockeye-d commented:
Finally got around to fixing this. I
* added back the Godot 3 keywords
* added the rest of Godot 4's Variant types
* replaced the built-in node name highlighting with a more generic one that just highlights all words that start with an uppercase letter
* Added a new highlighting rule for CONSTANT_CASE constants

Andriamanitra commented on runtime/syntax/gdscript.yaml:
```diff
@@ -8,18 +8,21 @@ rules:
+    - type: "\\b(AABB|Array|Basis|Callable|Color|Dictionary|NodePath|Object|PackedByteArray|PackedColorArray|PackedFloat32Array|PackedFloat64Array|PackedInt32Array|PackedInt64Array|PackedVector2Array|PackedVector3Array|PackedVector4Array|Plane|PoolByteArray|PoolColorArray|PoolIntArray|PoolRealArray|PoolVector2Array|PoolVector3Array|Projection|Quat|Quaternion|Rect2|Rect2i|RID|Signal|String|StringN ...
+    - constant: "\\b([A-Z][A-Z0-9_]*)\\b"
+    # Built-in node names
+    - identifier: "\\b([A-Z][a-zA-Z0-9_]*)\\b"
```
The definition for `identifier` is superset of `constant`, and because `identifier` is defined later in the file it will take precedence. Thus nothing is ever colored as `constant`. The comment is also wrong now as it matches all names, not just the built-in node names.

Andriamanitra reviewed (changes requested):
I tested with two GDScript files I found on Github:
* [syntax_test_gdscript.gd](https://github.com/beefsack/GDScript-sublime/blob/96f5dcf29728aa987123321e2544330eed991a3e/syntax_test_gdscript.gd)
* [long.gd](https://github.com/GDQuest/GDScript-formatter/blob/main/benchmarks/gdscript_files/long.gd)

Triple quoted strings are missing handling for escaped characters, and I think it would make sense to make triple quoted strings into strings rather than comments (the documentation no longer mentions triple quoted strings as a way to create comments).

Other than that everything seems to look good.

sockeye-d commented on runtime/syntax/gdscript.yaml:
```diff
@@ -7,19 +7,22 @@ rules:
     - statement.meta: "@[A-Za-z_][A-Za-z0-9_]*\\b"
     # Keywords
-    - statement: "\\b(and|as|assert|await|break|breakpoint|class|class_name|const|continue|elif|else|enum|export|extends|for|func|get|if|in|is|map|master|mastersync|match|not|onready|or|pass|remote|remotesync|return|self|set|setget|slave|slavesync|signal|sync|tool|var|while|yield)\\b"
+    - statement: "\\b(and|as|assert|await|break|breakpoint|class|class_name|const|continue|elif|else|enum|export|extends|for|func|get|if|in|is|map|master|mastersync|match|not|onready|or|pass|remote|remotesync|return|self|set|setget|slave|slavesync|signal|static|super|sync|tool|var|when|while|yield)\\b"
```
I think that `void` should be highlighted like a type instead of a keyword, for parity with Godot's built in syntax highlighting, is that okay?

Andriamanitra reviewed (commented):
The triple quoted string highlighting actually *needs* to be above single quoted strings or the highlighting will see `"""x"""` as three single quoted strings rather than one triple quoted string. [^1] Check with [syntax_test_gdscript.gd](https://github.com/beefsack/GDScript-sublime/blob/96f5dcf29728aa987123321e2544330eed991a3e/syntax_test_gdscript.gd) (it's easier to see what's going on while testing if you temporarily make triple quoted strings different color). You probably also need `skip: "\\\\."` to pass this test case:
```gdscript
var x = """ \""" """
var y = 123
```

[^1]: confusingly this is different than the situation we had earlier where the later rule was being applied on top of an earlier rule, that won't happen with groups with `start`/`end`.

sockeye-d commented:
I think this should fix that?

sockeye-d commented on runtime/syntax/gdscript.yaml:
```diff
@@ -7,38 +7,45 @@ rules:
     - constant: "\\b(INF|NAN|PI|TAU)\\b"
     - constant.bool: "\\b(null|true|false)\\b"
+    # Function calls
+    - identifier: "[a-zA-Z_0-9]+\\("
```
Yep, the opening bracket just gets rehighlighted elsewhere. I couldn't figure out how to get a lookahead working so that was my next best option. I've fixed the function highlighting to check for word boundaries and handle numbers properly

JoeKar replied:
But that is not more than an unintended hack.
This should stop working after the highlighter is reworked, in which it builds kind of a tree.
Unless we improve the `skip` in that way in which it really was intended to work (provide a sort of a look ahead/around).

BTW:
Missed this one `- identifier: "func\\s+[a-zA-Z_0-9]+"`?

Andriamanitra replied:
> But that is not more than an unintended hack.
> This should stop working after the highlighter is reworked, in which it builds kind of a tree.
> Unless we improve the skip in that way in which it really was intended to work (provide a sort of a look ahead/around).

I'm pretty sure it's an *intended* hack. It is already in use in other syntaxes [^1], some of which have been using it ever since the current highlighting system was introduced in 2fcb40d5.

[^1]: Grepping for `\\("` found kvlang, Lua, and PHP, and I'm pretty sure there are others that use the same trick for something slightly different
